rosbag2_test_common/include/rosbag2_test_common/publication_manager.hpp

```diff
@@ -103,10 +103,11 @@ class PublicationManager
   /// \param n_subscribers_to_match Number of subscribers publisher should have for match.
   /// \return true if find publisher by specified topic_name and publisher has specified number of
   ///   subscribers, otherwise false.
-  template<typename Timeout>
+  template<typename DurationRepT = int64_t, typename DurationT = std::milli>
   bool wait_for_matched(
     const char * topic_name,
-    Timeout timeout, size_t n_subscribers_to_match = 1)
+    std::chrono::duration<DurationRepT, DurationT> timeout = std::chrono::seconds(30),
```

Comment:
agree with @Karsten1987 , Do we actually need 30 seconds in default? If actually wait happens, this will take a long time to timeout every test, i think that possibly slows down CI / CD system.

Reply:
Okay. I change the default timeout to 10s as Karsten1987' suggestion.

Reply:
@emersonknapp I am not really sure how long is appropriate, would you mind sharing idea? CC: @Karsten1987 @Barry-Xu-2018

Reply:
10s seems fine - if all the tests are timing out, it will take a very long time no matter what, but at least 10s is 3 times faster than 30s :)